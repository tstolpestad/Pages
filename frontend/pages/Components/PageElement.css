.element {
	background: white;
	padding: 2px;
	width: 100%;
}
