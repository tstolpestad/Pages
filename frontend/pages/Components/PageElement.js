import React from "react";
import {Draggable} from "react-beautiful-dnd"
import classNames from './PageElement.css';


export const PageElement = ({element, index, dnd, }) => {
	//todo: styling
	return dnd ?
		<Draggable draggableId={element.id} index={index}>
			{
				provided => (
					<div className={classNames.element} {...provided.draggableProps} {...provided.dragHandleProps} ref={provided.innerRef}>
						{element.title}
					</div>
				)

			}
		</Draggable>
		:<div className={classNames.element}>
		{element.title}
	</div>
}
