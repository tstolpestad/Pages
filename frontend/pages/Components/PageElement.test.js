/**
 * Created by Tonje on 12/18/2018.
 */
import {PageElement} from './PageElement'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';



describe("PageElement", ()=> {
	const wrapper = shallow(<PageElement element="test"/>);
	it("should match snapshot, for regression testing", ()=>{
		expect(toJson(wrapper)).toMatchSnapshot();
	})
	it("should render an element with text", ()=>{
		console.log(toJson(wrapper))
		expect(toJson(wrapper)).toBe(true);
	})
	it("should be a droppable element", ()=> {

	})
})
