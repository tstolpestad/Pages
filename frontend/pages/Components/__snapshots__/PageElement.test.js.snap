// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageElement should match snapshot, for regression testing 1`] = `
<div
  className="element"
>
  test
</div>
`;
